CorePayments/src/main/java/com/paypal/android/corepayments/features/eligibility/CheckEligibilityResultListener.kt

```diff
@@ -0,0 +1,23 @@
+package com.paypal.android.corepayments.features.eligibility
+
+import com.paypal.android.corepayments.PayPalSDKError
+
+/**
+ * Implement this interface to receive results from calls to [EligibilityClient.check].
+ */
+interface CheckEligibilityResultListener {
+
+    /**
+     * Called when [EligibilityClient.check] succeeds.
+     *
+     * @param result The [EligibilityResult].
+     */
+    fun onCheckEligibilitySuccess(result: EligibilityResult)
```

Comment:
Not sure if PPCP has this pattern, but do we want to have a single function return a sealed class result?

Reply:
Yeah that's one of the first breaking changes we'll have to make in `v2`. The current pattern in `v1` is outdated and should change. For now, this makes since to keep consistent with the other listeners.

This pattern made since initially, but it gets cumbersome when browser switching is involved e.g. [ApproveOrderListener.kt](https://github.com/paypal/paypal-android/blob/7285f90d939a7896fbfc868a2390deb8202c1059/CardPayments/src/main/java/com/paypal/android/cardpayments/ApproveOrderListener.kt#L9).